filter/src/main/java/com/iluwatar/filter/App.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This project is licensed under the MIT license. Module model-view-viewmodel is using ZK framework licensed under LGPL (see lgpl-3.0.txt).
+ *
+ * The MIT License
+ * Copyright © 2014-2022 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.filter;
+
+import com.iluwatar.filter.criteria.AndCriteria;
+import com.iluwatar.filter.criteria.Criteria;
+import com.iluwatar.filter.criteria.InStockCriteria;
+import com.iluwatar.filter.criteria.MediumPriceCriteria;
+import com.iluwatar.filter.product.Product;
+import com.iluwatar.filter.product.ProductCategory;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Main class.
+ * Demonstrates how two filters can be used together using the and filter.
+ */
+public class App {
+  /**
+   * Program entry point.
+   * @param args runtime arguments.
+   */
+  public static void main(String[] args) {
+    Criteria<Product> inStockCriteria = new InStockCriteria();
```

Comment:
Use `var` for local variables, where possible

Reply:
Add comments. This is study material for most of the people, so it needs to be well explained.